service/src/main/kotlin/no/nav/su/se/bakover/service/utbetaling/UtbetalingService.kt

```diff
@@ -0,0 +1,85 @@
+package no.nav.su.se.bakover.service.utbetaling
+
+import arrow.core.Either
+import arrow.core.left
+import arrow.core.right
+import no.nav.su.se.bakover.database.ObjectRepo
+import no.nav.su.se.bakover.domain.Attestant
+import no.nav.su.se.bakover.domain.oppdrag.NyUtbetaling
+import no.nav.su.se.bakover.domain.oppdrag.Utbetaling
+import no.nav.su.se.bakover.domain.oppdrag.simulering.SimuleringClient
+import no.nav.su.se.bakover.domain.oppdrag.utbetaling.UtbetalingPublisher
+import no.nav.su.se.bakover.service.utbetaling.StartUtbetalingFeilet.FantIkkeSak
+import no.nav.su.se.bakover.service.utbetaling.StartUtbetalingFeilet.HarIngenOversendteUtbetalinger
+import no.nav.su.se.bakover.service.utbetaling.StartUtbetalingFeilet.SendingAvUtebetalingTilOppdragFeilet
+import no.nav.su.se.bakover.service.utbetaling.StartUtbetalingFeilet.SimuleringAvStartutbetalingFeilet
+import no.nav.su.se.bakover.service.utbetaling.StartUtbetalingFeilet.SisteUtbetalingErIkkeEnStansutbetaling
+import org.slf4j.LoggerFactory
+import java.util.UUID
+
+class UtbetalingService(
+    private val repo: ObjectRepo,
+    private val simuleringClient: SimuleringClient,
+    private val utbetalingPublisher: UtbetalingPublisher
+) {
+    private val log = LoggerFactory.getLogger(this::class.java)
+
+    fun startUtbetalinger(sakId: UUID): Either<StartUtbetalingFeilet, Utbetaling> {
+        val sak = repo.hentSak(sakId) ?: return FantIkkeSak.left()
+        val sistSendteUtbetaling =
+            sak.oppdrag.sisteOversendteUtbetaling() ?: return HarIngenOversendteUtbetalinger.left()
+        if (!sistSendteUtbetaling.erStansutbetaling()) return SisteUtbetalingErIkkeEnStansutbetaling.left()
```

Comment:
Eksisterer det en sperre som forhindrer at man setter i gang nye utbetalinger mens en stans er "underveis"?. Sannsynligvis ikke i scope for nå, men her kan det vel oppstå problemer i overgangen mellom stønadsperioder.. f.eks hvis neste periode behandles før utløpet av en periode som er stanset (utbetalingen er den siste, men gjelder for en annen periode enn utbetalingen som er stanset)

Reply:
Nei, den sperren mangler tror jeg. Det bør gjøres mer enhetlig. På samme måte som vi kan ha enhetlig validering av simulering og alltid gjøre en simulering rett før utbetalinga.

Reply:
Men blir nye tasks som du sier.